snapshot-labs/envelop
# feat: preview for email templates

When enabled (`PREVIEW=true` in .env), the preview function will:

- disable the actual email sending (no live email sent)
- render the email into a temp HTML file

There is an additional variable in .env (`AUTO_OPEN_PREVIEW`) which will auto open the tmp email file in your browser, when `send()` is called

## How to test 

Ensure the following variables presence in the .env file

```
PREVIEW=true
AUTO_OPEN_PREVIEW=true
```

Start the app as usual, emails preview will open in the browser instead of being actually sent.

### Notes

To avoid additional dependencies:

- tmp file creation code is generic and basic, and may no works in all env
- tmp file auto open in browser is also generic, and should works in most of env, but not as robust as with a real NPM package.
